Content.Shared/ImmovableRod/ImmovableRodSystem.cs

```diff
@@ -0,0 +1,161 @@
+        }
+    }
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ImmovableRodComponent, StartCollideEvent>(OnCollide);
+        SubscribeLocalEvent<ImmovableRodComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<ImmovableRodComponent, ExaminedEvent>(OnExamined);
+    }
+
+    private void OnMapInit(EntityUid uid, ImmovableRodComponent component, MapInitEvent args)
+    {
+        if (!EntityManager.TryGetComponent(uid, out PhysicsComponent? phys))
+            return;
+
+        _physics.SetLinearDamping(uid, phys, 0f);
+        _physics.SetFriction(uid, phys, 0f);
+        _physics.SetBodyStatus(uid, phys, BodyStatus.InAir);
+
+        var xform = Transform(uid);
+        var (worldPos, worldRot) = _transform.GetWorldPositionRotation(uid);
+        var vel = worldRot.ToWorldVec() * component.MaxSpeed;
+
+        if (component.RandomizeVelocity)
+        {
+            vel = component.DirectionOverride.Degrees switch
+            {
+                0f => _random.NextVector2(component.MinSpeed, component.MaxSpeed),
+                _ => worldRot.RotateVec(component.DirectionOverride.ToVec()) * _random.NextFloat(component.MinSpeed, component.MaxSpeed)
+            };
+        }
+
+        _physics.ApplyLinearImpulse(uid, vel, body: phys);
+        xform.LocalRotation = (vel - worldPos).ToWorldAngle() + MathHelper.PiOver2;
+    }
+
+    protected virtual void OnCollide(EntityUid uid, ImmovableRodComponent component, ref StartCollideEvent args)
+    {
+        var ent = args.OtherEntity;
+
+        // Clang!
+        if (_random.Prob(component.HitSoundProbability))
+            _audio.PlayPvs(component.Sound, uid);
+
+        // Oh nyo.
+        if (HasComp<ImmovableRodComponent>(ent))
+            HandleRodRodCollision(uid, ent);
+        // Mobs never automatically get deleted by Rodney, because being instantly deleted is not any fun.
+        else if (TryComp<MobStateComponent>(ent, out var mob))
+            HandleRodMobCollision(uid, component, args);
+        // Admiral Roddington XIV may have been limited in his desire to delete everythng.
+        // What's the point of an immovable rod if it deletes all those organs it's just splattered?
+        else if (component.ShouldDelete && !_whitelist.IsWhitelistPass(component.DeletionBlacklist, ent))
+            QueueDel(ent);
+        // The Amulet of Yendor has powers some may deem... unnatural. Clang!
+        else if (component.Damage != null)
+            _damageable.TryChangeDamage(ent, component.Damage, ignoreResistances: true);
+    }
+
+    private void HandleRodRodCollision(EntityUid uid, EntityUid ent)
+    {
+        // :panik:
+        var coords = Transform(uid).Coordinates;
+        _popup.PopupCoordinates(Loc.GetString("immovable-rod-collided-rod-not-good"), coords, PopupType.LargeCaution);
+
+        // :kalm:
+        Del(uid);
+        Del(ent);
+
+        // :panik:
+        Spawn("Singularity", coords);
```

Comment:
This at least needs to be `SpawnAtPosition` and may need to be called before the deletion.